Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/util/UiMatcher.java

```diff
@@ -426,12 +427,19 @@ protected boolean matchesSafely(View target) {
 					return false;
 				}
 
-				Bitmap expectedBitmap = ((BitmapDrawable) expectedDrawable).getBitmap();
+				Bitmap expectedBitmap;
+
 				if (targetDrawable instanceof BitmapDrawable) {
 					Bitmap targetBitmap = ((BitmapDrawable) targetDrawable).getBitmap();
+					expectedBitmap = ((BitmapDrawable) expectedDrawable).getBitmap();
 					return targetBitmap.sameAs(expectedBitmap);
 				} else if (targetDrawable instanceof StateListDrawable) {
 					Bitmap targetBitmap = ((BitmapDrawable) targetDrawable.getCurrent()).getBitmap();
+					expectedBitmap = ((BitmapDrawable) expectedDrawable).getBitmap();
+					return targetBitmap.sameAs(expectedBitmap);
+				} else if (targetDrawable instanceof VectorDrawable) {
+					Bitmap targetBitmap = vectorToBitmap((VectorDrawable) expectedDrawable);
+					expectedBitmap = vectorToBitmap((VectorDrawable) expectedDrawable);
```

Comment:
Is there a way to compare vector drawables without rasterizing them?

Reply:
the only other way i found so far was settings tags and comparing them. But tags may be set wrong, so that won't be a proper solution